cmd/goldsmith/ui/ui.go

```diff
@@ -8,7 +8,35 @@ import (
 
 // FFTStreamer buffers a streamer and also computes an FFT whos chunks are available on [FFTChan].
 type FFTStreamer interface {
-	beep.Streamer
+	beep.Streamer // Are the methods from this being used? if not I'd leave it out
+
+	// 1) Could FFTUpdateSignal and FFTChan be consolidated into a single
```

Comment:
No because they have different responsibilities!  One reads in the data to perform the FFT on (as fast as it's buffer fills).  The other one determines *when* it is appropriate to update to the next frame/fft that has been calculated